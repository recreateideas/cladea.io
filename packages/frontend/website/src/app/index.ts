export * from "./App";
export * from "./features";
