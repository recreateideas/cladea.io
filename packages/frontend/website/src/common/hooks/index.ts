export * from "./ClickOutside";

export * from "./use-user-agent";
export * from "./use-scroll-y";
export * from "./use-margin";
export * from "./use-breakpoints";
