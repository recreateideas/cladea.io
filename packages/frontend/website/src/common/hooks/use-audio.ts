import { useState, useEffect } from "react";

interface Props {
  sound: string;
  loop?: boolean;
}
export const useAudio = (props: Props) => {
  const { sound, loop } = props;
  const [audio] = useState(new Audio(sound));
  const [isPlaying, setIsPlaying] = useState(false);

  const toggle = (is?: boolean) => {
    console.log(isPlaying);
    setIsPlaying(is !== undefined ? is : !isPlaying);
  };

  useEffect(() => {
    console.log(isPlaying);
    isPlaying ? audio.play() : audio.pause();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isPlaying]);

  useEffect(() => {
    audio.addEventListener("ended", () => setIsPlaying(!!loop));
    return () => {
      audio.removeEventListener("ended", () => setIsPlaying(!!loop));
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return [toggle];
};
