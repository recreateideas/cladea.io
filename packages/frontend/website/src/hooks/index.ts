export * from "./ClickOutside";
export * from "./use-scroll-y";
export * from "./use-margin";
export * from "./use-breakpoints";
