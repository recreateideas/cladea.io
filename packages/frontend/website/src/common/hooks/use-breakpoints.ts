import { useState, useLayoutEffect } from "react";
import { Breakpoints, selectors, useSelector } from "src/redux";

const sortBreakpoints = (breakpoints: Breakpoints) =>
  Object.keys(breakpoints).sort((a, b) => breakpoints[a] - breakpoints[b]);

const getBreakpointFromWidth = (width: number, breakpoints: Breakpoints) => {
  const sorted = sortBreakpoints(breakpoints);
  let [matching] = sorted;
  sorted.forEach((breakpoint) => {
    if (width > breakpoints[breakpoint]) {
      matching = breakpoint;
    }
  });
  return matching;
};

export const useBreakpoints = () => {
  const isClient = typeof window === "object";
  const getWidth = () => (isClient ? window.innerWidth : 0);
  const { theme: themeSelectors } = selectors;
  const breakpoints = useSelector(themeSelectors.breakpoints);
  const startBreakpoint = getBreakpointFromWidth(getWidth(), breakpoints);
  const [breakpoint, setBreakpoint] = useState(startBreakpoint);
  useLayoutEffect(() => {
    if (!isClient) {
      return;
    }
    function handleResize() {
      const newWidth = getWidth();
      const newBreakpoint = getBreakpointFromWidth(newWidth, breakpoints);
      if (newBreakpoint !== breakpoint) {
        setBreakpoint(newBreakpoint);
      }
    }
    window.addEventListener("resize", handleResize);
    return () => window.removeEventListener("resize", handleResize);
    // eslint-disable-next-line
  }, [startBreakpoint]);
  return breakpoint;
};
