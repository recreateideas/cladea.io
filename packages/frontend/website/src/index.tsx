import "./app/ui-core/theme/GlobalStyle/roboto.css";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { store } from "./redux";
import { ServicedApp, ThemeProviderConnected } from "./app";

ReactDOM.render(
  <Provider store={store}>
    <ThemeProviderConnected>
      <ServicedApp />
    </ThemeProviderConnected>
  </Provider>,
  document.getElementById("root")
);
