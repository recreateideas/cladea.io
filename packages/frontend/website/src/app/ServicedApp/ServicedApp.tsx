import { ReactElement } from 'react';
import { useHealthCheck, useServicePorts } from '../../hooks';
import { useTheme } from 'styled-components';
import { Loader } from '../ui-core';
import { App } from '../App';
import { Theme } from '@types';

export const ServicedApp = (): ReactElement => {
    const servicePorts = useServicePorts();
    const isHealthy = useHealthCheck({ servicePorts });
    const isChecking = isHealthy === undefined;
    const theme = useTheme() as Theme;
    return (
        <div id="app">
            {isHealthy ? (
                <App />
            ) : !servicePorts ? (
                <Loader
                    {...{
                        className: 'setting-up',
                        color: theme.dsl.palette.fonts.success,
                        message: 'setting up services...',
                    }}
                />
            ) : isChecking ? (
                <Loader
                    {...{
                        className: 'checking',
                        color: theme.dsl.palette.brand.primary[500],
                        message: 'checking services...',
                    }}
                />
            ) : (
                <Loader
                    {...{
                        className: 'not-responding',
                        color: theme.dsl.palette.fonts.error,
                        freeze: true,
                        message: 'Services are not responding. Cmd + R to retry.',
                    }}
                />
            )}
        </div>
    );
};
