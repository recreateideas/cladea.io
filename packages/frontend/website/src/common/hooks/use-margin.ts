import { selectors, useSelector } from "src/redux";
import { useBreakpoints } from "./use-breakpoints";

export const useMargin = () => {
  const { theme: themeSelectors } = selectors;
  const margins = useSelector(themeSelectors.margins);
  const breakpoint = useBreakpoints();
  return margins[breakpoint];
};
