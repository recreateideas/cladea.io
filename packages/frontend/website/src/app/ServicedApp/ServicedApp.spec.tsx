/* eslint-disable import/first */
import { ServicedApp } from './ServicedApp';
jest.mock('../../hooks');
import * as hooks from '../../hooks';
import { TestProvider } from 'src/tests';

const { fullMount } = TestProvider();

describe('<ServicedApp/>', () => {
    it('should render', () => {
        const root = fullMount(<ServicedApp />);
        expect(root).toMatchSnapshot();
    });
    it('should render .setting-up loader when service-ports are undefined', () => {
        jest.spyOn(hooks, 'useServicePorts').mockReturnValue(undefined);
        const wrapper = fullMount(<ServicedApp />);
        const Loader = wrapper.exists('.setting-up');
        expect(Loader).toBeTruthy();
    });

    it('should render App when isHealthy is true', () => {
        jest.spyOn(hooks, 'useServicePorts').mockReturnValue({ 'some-service': 1111 });
        jest.spyOn(hooks, 'useHealthCheck').mockReturnValue(true);
        const wrapper = fullMount(<ServicedApp />);
        const Loader = wrapper.exists('App');
        expect(Loader).toBeTruthy();
    });
    it('should render .checking loader when isHealthy is undefined', () => {
        jest.spyOn(hooks, 'useServicePorts').mockReturnValue({ 'some-service': 1111 });
        jest.spyOn(hooks, 'useHealthCheck').mockReturnValue(undefined);
        const wrapper = fullMount(<ServicedApp />);
        const Loader = wrapper.exists('.checking');
        expect(Loader).toBeTruthy();
    });
    it('should render .not-responding loader when service-ports are loaded and isHealthy is false', () => {
        jest.spyOn(hooks, 'useServicePorts').mockReturnValue({ 'some-service': 1111 });
        jest.spyOn(hooks, 'useHealthCheck').mockReturnValue(false);
        const wrapper = fullMount(<ServicedApp />);
        const Loader = wrapper.exists('.not-responding');
        expect(Loader).toBeTruthy();
    });
});
