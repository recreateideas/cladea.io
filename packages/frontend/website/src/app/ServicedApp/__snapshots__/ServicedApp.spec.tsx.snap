// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServicedApp/> should render 1`] = `
Array [
  .c0 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .react-spinner-material {
  margin: auto;
}

.c1 {
  height: 100%;
  width: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  top: calc(50% + 40px);
}

<div
    id="app"
  >
    <div
      className="c0 loader setting-up"
    >
      <div
        className="react-spinner-material"
        style={
          Object {
            "borderColor": "#0FC9A5",
            "borderWidth": 3,
            "height": 40,
            "width": 40,
          }
        }
      />
      <div
        className="c1"
      >
        setting up services...
      </div>
    </div>
  </div>,
  "",
]
`;
